framework/OutStreams/OutStreamInterface.py

```diff
@@ -0,0 +1,117 @@
+
+class OutStreamInterface(BaseInterface):
+  """
+    Base class for other OutStream Interfaces (Print, Plot).
+    Not meant to be an interface itself.
+  """
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for class "cls".
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for specifying the input of cls.
+    """
+    spec = super().getInputSpecification()
+    spec.addParam('dir', descr=r"""Designates the directory in which files should be saved. If not
+        an absolute directory, assumes relative to the WorkingDir. """)
+    spec.addParam('overwrite', param_type=InputTypes.BoolType, descr=r"""If True, then any conflicting
+        existing files will be overwritten.""")
+    spec.addSub(InputData.parameterInputFactory('filename', contentType=InputTypes.StringType,
+        descr=r"""Sets the name to use for the saved file."""))
+    return spec
+
+  def __init__(self):
+    """
+      Init of Base class
+      @ In, None
+      @ Out, None
+    """
+    super().__init__()
+    self.printTag = 'OutStreamInterface'
+    self.overwrite = True       # overwrite existing creations?
+    self.subDirectory = None    # directory to save generated files to
+    self.filename = ''          # target file name
+    self.numberAggregatedOS = 1 # number of aggregated outstreams # no addl info from original OutStream
+
+  def handleInput(self, spec):
+    """
+      Loads the input specs for this object.
+      @ In, spec, InputData.ParameterInput, input specifications
+      @ Out, None
+    """
+    super().handleInput(spec)
+    self.subDirectory = spec.parameterValues.get('dir')
+    self.overwrite = spec.parameterValues.get('overwrite')
+    fname = spec.findFirst('filename')
+    if fname is not None:
+      self.filename = fname
+
+  def initialize(self, stepEntities):
+    """
+      Initialize for a new Step
+      @ In, stepEntities, dict, Entities from the Step
+      @ Out, None
+    """
+    super().initialize()
+    if self.subDirectory is not None:
+      if not os.path.exists(self.subDirectory):
+        os.makedirs(self.subDirectory)
+
+  # this method is used by the original OutStreams, GeneralPlot and File Print.
+  def legacyCollectSources(self, inDict):
+    """
+      Collect the usable sources in the format this plotter expects.
+      @ In, inDict, dict, Step entities
+      @ Out, None
+    """
+    self.sourceData = []
+    for outIndex in range(self.numberAggregatedOS):
+      foundData = False
+      for output in inDict['Output']:
+        if output.name.strip() == self.sourceName[outIndex] and output.type in DataObjects.factory.knownTypes():
+          self.sourceData.append(output)
+          foundData = True
+          break
+      if not foundData:
+        for inp in inDict['Input']:
+          if not type(inp) == type(""):
```

Comment:
I believe you can just use `str` instead of `type("")`.  Might make it more clear exactly what the conditional is doing.

Reply:
this is inherited code from the old OutStreamManager, but I can make that change.